third_party/accessibility/ax/platform/ax_platform_node_mac.mm

```diff
@@ -699,6 +710,10 @@ - (NSValue*)AXSelectedTextRangeInternal {
     start = _node->GetIntAttribute(ax::mojom::IntAttribute::kTextSelStart);
     end = _node->GetIntAttribute(ax::mojom::IntAttribute::kTextSelEnd);
   }
+  if (start == -1 && end == -1) {
+    return [NSValue valueWithRange:{0, 0}];
+  }
+  NSAssert(start >= 0 && end >= 0, @"selection is invalid");
```

Comment:
You could move this check right above the preceding block and check for both >= 0 or both < 0 to declare the invariant you expect at the first moment it should hold, then change this check to an `||` bailout. In practice, it probably doesn't matter since IIRC we don't set `NS_BLOCK_ASSERTIONS` in our release build config.